### PR TITLE
docs: Document our conventions for experimental features

### PR DESCRIPTION
**Pull request by odzhychko**
Fixes: MODELIX-547

### CODE REVIEW

File: docs/modules/ROOT/pages/main/development.adoc [hunk 1]
```diff
@@ -23,4 +23,17 @@ We aim to provide a Spring release `R1` and an Autumn release `R2`.
...
+To gain an overview over existing components, please refer to the list of components xref:reference/repositories-components.adoc[here].
+
+[#experimental_features]
+== Experimental features
+
+Sometimes we want to implement a feature or expose an API that is experimental.
+
+When doing so, we mark the APIs as experimental.
+In TypeScript we use the https://tsdoc.org/pages/tags/experimental/[documentation tag `@experimental`].
+For Kotlin we create the annotation `@UnstableModelixFeature` for an https://kotlinlang.org/docs/opt-in-requirements.html[Opt-in requirement]
```
Comment (languitar) [inline docs/modules/ROOT/pages/main/development.adoc]:
```suggestion
For Kotlin, we use the annotation `@UnstableModelixFeature` for an https://kotlinlang.org/docs/opt-in-requirements.html[opt-in requirement]
```

That annotation was already created.

File: docs/modules/ROOT/pages/main/development.adoc [hunk 2]
```diff
@@ -23,4 +23,17 @@ We aim to provide a Spring release `R1` and an Autumn release `R2`.
...
+
+When marking something as experimental, we document the reason for it being experimental and when we will make it stable or remove it.
```
Comment (languitar) [inline docs/modules/ROOT/pages/main/development.adoc]:
```suggestion
When marking something as experimental, we document the reason for it being experimental and our plans for stabilizing the feature.
```

File: docs/modules/ROOT/pages/main/development.adoc [hunk 3]
```diff
@@ -23,4 +23,17 @@ We aim to provide a Spring release `R1` and an Autumn release `R2`.
...
+
+An example in our Kotlin code can be found https://github.com/modelix/modelix.core/blob/e14d15f/model-client/src/jsMain/kotlin/org/modelix/model/client2/ClientJS.kt[here]. An example in TypeScript can be found https://github.com/modelix/modelix.core/blob/main/vue-model-api/src/useRootNode.ts[here].
```
Comment (languitar) [inline docs/modules/ROOT/pages/main/development.adoc]:
```suggestion
An example in our Kotlin code can be found https://github.com/modelix/modelix.core/blob/e14d15f/model-client/src/jsMain/kotlin/org/modelix/model/client2/ClientJS.kt[here].
An example in TypeScript can be found https://github.com/modelix/modelix.core/blob/main/vue-model-api/src/useRootNode.ts[here].
```

**Review by languitar — APPROVED** [review]
Just some rewording ideas.

**Comment by languitar** [comment]
Btw, for the documentation, it's a good idea to configure [language tool](https://languagetool.org/) in your editor if not done yet. There's an LSP for that. Language tool has an option to configure your native tongue. This helps finding issues German speakers would typically do.

**Comment by odzhychko** [comment]
@languitar Thanks for the rewording, I applied them.

**Comment by languitar** [comment]
Sorry, missed that ping

**Review by languitar — APPROVED** [review]

**Pull request merged**